presto-tests/src/main/java/com/facebook/presto/tests/AbstractTestQueries.java

```diff
@@ -2345,111 +2346,220 @@ public void testJoinWithNullValues()
     public void testJoinWithMultipleInSubqueryClauses()
             throws Exception
     {
-        String queryPrefix = "SELECT * " +
+        Function<String, String> queryPrefix = (String joinType) -> "SELECT * " +
                 "FROM " +
-                "    (VALUES 1,2,3) t(x) " +
-                "  JOIN " +
-                "    (VALUES 1,2,3) t2(y) " +
+                "    (VALUES 1,2,3,4) t(x) " +
+                joinType + " JOIN " +
+                "    (VALUES 1,2,3,5) t2(y) " +
                 "  ON ";
 
+        String twoDuplicatedInSubqueriesCondition = "(x in (VALUES 1,2,3)) = (y in (VALUES 1,2,3)) AND (x in (VALUES 1,2,4)) = (y in (VALUES 1,2,4))";
         assertQuery(
-                queryPrefix + "(x in (VALUES 1,2,3)) = (y in (VALUES 1,2,3)) AND (x in (VALUES 1,2)) = (y in (VALUES 1,2))",
+                queryPrefix.apply("") + twoDuplicatedInSubqueriesCondition,
                 "VALUES (1,1), (1,2), (2,2), (2,1), (3,3)");
         assertQuery(
-                queryPrefix + "(x in (VALUES 1,2)) = (y in (VALUES 1,2)) AND (x in (VALUES 1)) = (y in (VALUES 3))",
-                "VALUES (2,2), (2,1)");
+                queryPrefix.apply("")
+                        + "(x in (VALUES 1,2)) = (y in (VALUES 1,2)) AND (x in (VALUES 1)) = (y in (VALUES 3))",
+                "VALUES (2,2), (2,1), (3,5), (4,5)");
+        assertQuery(
+                queryPrefix.apply("")
+                        + "(x in (VALUES 1,2)) = (y in (VALUES 1,2)) AND (x in (VALUES 1)) != (y in (VALUES 3))",
+                "VALUES (1,2), (1,1), (3, 3), (4,3)");
+        assertQuery(
+                queryPrefix.apply("")
+                        + "(x in (VALUES 1)) = (y in (VALUES 1)) AND (x in (SELECT 2)) != (y in (SELECT 2))",
+                "VALUES (2,3), (2,5), (3, 2), (4,2)");
+
+        assertQueryFails(
+                queryPrefix.apply("left") + "x in (VALUES 1)",
+                ".*IN with subquery predicate in join condition not supported");
+        assertQueryFails(
+                queryPrefix.apply("left") + "y in (VALUES 1)",
+                ".*IN with subquery predicate in join condition not supported");
         assertQuery(
-                queryPrefix + "(x in (VALUES 1,2)) = (y in (VALUES 1,2)) AND (x in (VALUES 1)) != (y in (VALUES 3))",
-                "VALUES (1,2), (1,1), (3, 3)");
+                queryPrefix.apply("left") + twoDuplicatedInSubqueriesCondition,
+                "VALUES (1,1), (1,2), (2,2), (2,1), (3,3), (4, null)");
+        assertQueryFails(
```

Comment:
I applied your comment, but I am not a very proud of that. I am not sure of its value.